hindsight/input/heka_s3.lua

```diff
@@ -0,0 +1,91 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+--[[
+## Reader for the S3 Heka files in compressed or uncompressed form
+
+Retrieves/reads each file from the `s3_file_list`. The primary use of this
+plugin is to feed the transformed/validated data into analysis plugins.
+
+### Sample Configuration
+```lua
+filename        = "heka_s3.lua"
+s3_bucket       = "net-mozaws-prod-us-west-2-pipeline-data"
+s3_file_list    = "files.ls.1"
+tmp_dir         = "/mnt/work/tmp"
+```
+--]]
+
+require "io"
+require "heka_stream_reader"
+require "os"
+require "string"
+
+local tmp_dir       = read_config("tmp_dir")
+local s3_bucket     = read_config("s3_bucket") or error("s3_bucket must be set")
+local logger        = read_config("Logger")
+local s3_file_list  = assert(io.open(read_config("s3_file_list")))
+
+
+local function process_file(hsr, fn)
+    local fh, err = io.open(fn)
+    if not fh then
+        print("failed to open", fn)
+        return
+    end
+
+    local found, consumed, read
+    repeat
+        repeat
+            found, consumed, read = hsr:find_message(fh)
+            if found then
+                inject_message(hsr)
+            end
+        until not found
+    until read == 0
+    fh:close()
+end
+
+
+local function execute_cmd(cmd, retries)
+    local rv = 1
+    for i=1, retries do
+        rv = os.execute(cmd)
+        if rv == 0 then
+            break
+        end
+    end
+    return rv
+end
+
+
+function process_message()
+    local hsr  = heka_stream_reader.new("s3")
+
+    for fn in s3_file_list:lines() do
+        local cmd
+        local tfn = string.format("%s/%s", tmp_dir, logger)
+        local ext = fn:match("%.([^.]-)$")
+        if ext == "zst" then
+            cmd = string.format("aws s3 cp s3://%s/%s - | zstd -d -c - > %s", s3_bucket, fn, tfn)
```

Comment:
in a generic sense I would be +1 but are in full control of these objects name and if we create anything that needs escaping it is a bug (I wouldn't be opposed to adding this if we refactor out a bootstrap module)